db/version_set.cc

```diff
@@ -6704,15 +6704,46 @@ uint64_t VersionSet::GetTotalBlobFileSize(Version* dummy_versions) {
   return all_versions_blob_file_size;
 }
 
-Status VersionSet::VerifyFileMetadata(const std::string& fpath,
-                                      const FileMetaData& meta) const {
+Status VersionSet::VerifyFileMetadata(ColumnFamilyData* cfd,
+                                      const std::string& fpath, int level,
+                                      FileMetaData& meta) {
```

Comment:
The FileMetaData is passed by reference (to mutable) and is left in an inconsistent state (released `table_reader_handle`). This makes it a bad internal API. It would be better to have only this function copy the FileMetaData rather than MaybeCreateVersion. I doesn't look like there's any intention to have postcondition changes to FileMetaData (such as leaving the file open) so it's better for separation of concerns to keep 'const' on all these function signatures you've updated.